packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -134,6 +134,127 @@ def test_metrics_agents_statsd(dcos_api_session):
             get_metrics_prom(dcos_api_session, agent, expected_metrics)
 
 
+@retrying.retry(wait_fixed=2000, stop_max_delay=300 * 1000)
+def check_metrics_prom(dcos_api_session, node, check_func):
+    """Get metrics from prometheus port on node and run check_func function,
+    asserting that it returns True.
+
+    Retries on non-200 status or failed assertions
+    for up to 300 seconds.
+
+    """
+    response = dcos_api_session.session.request(
+        'GET', 'http://{}:61091/metrics'.format(node))
+    assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
+    assert check_func(response) is True
```

Comment:
The only difference between `check_metrics_prom()` and `get_metrics_prom()` is the assertion after retrieving metrics from the Prometheus endpoint -- the request and timeout logic is duplicated. Let's refactor `get_metrics_prom()` so it only gets and returns the metrics response (while still asserting it's a 200), reuse that where `check_metrics_prom()` is used, and leave it up to test functions to assert whatever they need to in the response that gets returned.